payments/payment-methods/adyen/src/main/java/com/appcoins/payment_method/adyen/CreditCardTransaction.kt

```diff
@@ -20,9 +26,33 @@ class CreditCardTransaction internal constructor(
   val paymentResponse: PaymentResponse?
     get() = this._paymentResponse
 
-  private val _status = MutableStateFlow(initialStatus)
+  private val _status = MutableStateFlow(currentStatus)
 
-  override val status: Flow<TransactionStatus> = _status
+  override val status: Flow<TransactionStatus> = merge(
```

Comment:
The paypal transaction and this one should instantiate the flow in the same way. Maybe the paypal way is better to guarantee the order of emits.